lab-trent/lib/resourcemanager.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const fs = require('fs');
+
+const DATA_PATH = './data';
+
+let resourceCache = { };
+
+exports.addResource = function(resource) {
+  resourceCache[resource.id] = resource;
+  fs.writeFile(DATA_PATH + '/' + resource.id + '.json', JSON.stringify(resource), function(err) {
+    if (err) throw err;
+  });
+};
+
+exports.deleteResource = function(resource) {
+  delete resourceCache[resource.id];
```

Comment:
This doesn't actually remove the file from the directory structure, so you'll want to handle that with either `fs.unlink()` or another module like `del`